sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1968,7 +1986,7 @@ class Analyzer(override val catalogManager: CatalogManager)
     resolveExpression(
       expr,
       resolveColumnByName = nameParts => {
-        plan.resolve(nameParts, resolver)
+        plan.resolve(nameParts, resolver, withMetadata = false)
```

Comment:
This change fixes the UDF ordering issue.

Reply:
This change is really hard to understand. Why it's only in `resolveExpressionByPlanOutput` but not `resolveExpressionByPlanChildren`? How does it resolve the conflicts?

Reply:
I was thinking about this too - I can make a symmetric parameter for `resolveExpressionByPlanChildren`. This change likely also has a wider breadth than needed; primarily, we don't want `ResolveMissingReferences` to touch metadata as that should be resolved in `AddMetadataCol`.